spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MaskHead.scala

```diff
@@ -0,0 +1,175 @@
+
+    Graph(Array(features, proposals, labels), Array(maskFeatures, result))
+  }
+
+  private[nn] def maskPredictor(inChannels: Int,
+                                numClasses: Int,
+                                dimReduced: Int): Module[Float] = {
+    val convMask = SpatialFullConvolution(inChannels, dimReduced,
+      kW = 2, kH = 2, dW = 2, dH = 2)
+    val maskLogits = SpatialConvolution(nInputPlane = dimReduced,
+      nOutputPlane = numClasses, kernelW = 1, kernelH = 1, strideH = 1, strideW = 1)
+
+    // init weight & bias, Caffe2 implementation uses MSRAFill,
+    convMask.setInitMethod(MsraFiller(false), Zeros)
+    maskLogits.setInitMethod(MsraFiller(false), Zeros)
+
+    val model = Sequential[Float]()
+    model.add(convMask).add(ReLU[Float]()).add(maskLogits)
+    model
+  }
+
+  private[nn] def maskFeatureExtractor(inChannels: Int,
+                                       resolution: Int,
+                                       scales: Array[Float],
+                                       samplingRatio: Float,
+                                       layers: Array[Int],
+                                       dilation: Int,
+                                       useGn: Boolean = false): Module[Float] = {
+
+    require(dilation == 1, s"Only support dilation = 1, but get ${dilation}")
+
+    val model = Sequential[Float]()
+    model.add(Pooler(resolution, scales, samplingRatio.toInt))
+
+    var nextFeatures = inChannels
+    var i = 0
+    while (i < layers.length) {
+      val features = layers(i)
+      // todo: not support dilation convolution
+      val module = SpatialConvolution[Float](
+        nextFeatures,
+        features,
+        kernelW = 3,
+        kernelH = 3,
+        strideW = 1,
+        strideH = 1,
+        padW = dilation,
+        padH = dilation,
+        withBias = if (useGn) false else true
+      ).setName(s"mask_fcn{${i}}")
+
+      // weight init
+      module.setInitMethod(MsraFiller(false), Zeros)
+      model.add(module)
+      nextFeatures = features
+      i += 1
+    }
+    model.add(ReLU[Float]())
+  }
+}
+
+private[nn] class MaskPostProcessor(implicit ev: TensorNumeric[Float])
+  extends AbstractModule[Table, Tensor[Float], Float] {
+
+  @transient var rangeBuffer: Tensor[Float] = null
+  private val sigmoid = Sigmoid[Float]()
+
+  /**
+    * @param input feature-maps from possibly several levels, proposal boxes and labels
+    * @return the predicted boxlists are returned with the `mask` field set
+    */
+  override def updateOutput(input: Table): Tensor[Float] = {
+    val maskLogits = input[Tensor[Float]](1)
+    val bbox = input[Tensor[Float]](2) // N * 4
+    val labels = input[Tensor[Float]](3)
+
+    val num_masks = maskLogits.size(1)
+    if (rangeBuffer == null || rangeBuffer.nElement() != num_masks) {
+      rangeBuffer = Tensor[Float](num_masks)
+      TransformerOperation.initRangeTensor(num_masks, rangeBuffer)
```

Comment:
Do we need to rename the `TransformerOperation` since it's used here as well ?

Reply:
No, maybe we should add `range` method to Tensor which can generate tensor values in a range.

Reply:
then do we need to implement it for now ?

Reply:
OK, I will implement it in another pr.

Reply:
I just found `range` has been implemented, so I will just refactor code for this line.